parser.go

```diff
@@ -524,15 +529,23 @@ func (p *parser) parseEntryPoint() error {
 	return nil
 }
 
-func parseTags(comment string) (*TagDefinition, error) {
+func parseTags(comment string, tagGroups map[string][]string, tagGroupKeys *[]string) (*TagDefinition, error) {
```

Comment:
This function is getting difficult to understand - can we reformat or comment to make the conditions clear?

Reply:
Added a comment for the conditions - agreed we can refactor but I think lets take that out of scope for here.